### PR TITLE
[MRG+1] Normalize integer dimensions

### PR DESCRIPTION
**Pull request by MechCoder**
We should normalize integer dimensions as well because the isotropic GP hyper-parameters might be highly dominated by these if they are on a completely different scale from the others such as from [1, 47]

### CODE REVIEW

File: skopt/space/transformers.py [hunk 1]
```diff
@@ -120,7 +121,10 @@ def inverse_transform(self, X):
             raise ValueError("All values should be less than 1.0")
         if np.any(X < 0.0):
             raise ValueError("All values should be greater than 0.0")
-        return X * (self.high - self.low) + self.low
+        X_norm = X * (self.high - self.low) + self.low
```
Comment (glouppe) [inline skopt/space/transformers.py]: Isnt that the unnormalized `X`?

**Comment by glouppe** [comment]
Except this small comment, +1

File: skopt/space/transformers.py [hunk 2]
```diff
@@ -120,7 +121,10 @@ def inverse_transform(self, X):
...
+        if self.is_int:
+            return np.round(X_norm)
```
Comment (betatim) [inline skopt/space/transformers.py]: Should we convert the dtype as well? `np.round` returns an array of floats :-/

**Comment by MechCoder** [comment]
done

**Comment by betatim** [comment]
🎉

**Pull request merged**